tests/testsuite/env.rs

```diff
@@ -772,7 +773,9 @@ mod unix_test {
             || {
                 let result = Environment::default().collect();
 
-                assert!(matches!(result, Err(ConfigError::Foreign(_))));
+                assert!(
+                    matches!(result, Err(ConfigError::Foreign(e)) if e.downcast_ref::<InvalidUnicode>().is_some())
```

Comment:
We should have snapshot tests of the rendered error so we can validate how this fully looks to the user

Reply:
Done